src/WebJobs.Script.WebHost/Handlers/ExceptionProcessingHandler.cs

```diff
@@ -0,0 +1,143 @@
+using Microsoft.Azure.WebJobs.Host.Diagnostics;
+using Microsoft.Azure.WebJobs.Script.WebHost.Models;
+using Newtonsoft.Json;
+using ExceptionProcessor = System.Action<System.Web.Http.ExceptionHandling.ExceptionContext,
+    Microsoft.Azure.WebJobs.Script.AuthorizationLevel, Microsoft.Azure.WebJobs.Script.WebHost.Models.ApiErrorModel>;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Controllers
+{
+    public class ExceptionProcessingHandler : ExceptionHandler
+    {
+        private readonly IDictionary<Type, ExceptionProcessor> _handlers;
+        private readonly HttpConfiguration _config;
+        private readonly Lazy<TraceWriter> _traceWriterLoader;
+
+        public ExceptionProcessingHandler(HttpConfiguration config)
+        {
+            if (config == null)
+            {
+                throw new ArgumentNullException("config");
+            }
+
+            _config = config;
+            _traceWriterLoader = new Lazy<TraceWriter>(() => _config.DependencyResolver.GetService<TraceWriter>());
+            _handlers = InitializeExceptionHandlers();
+        }
+
+        private static IDictionary<Type, ExceptionProcessor> InitializeExceptionHandlers()
+        {
+            var handlers = new Dictionary<Type, ExceptionProcessor>
+            {
+                { typeof(CryptographicException), CryptographicExceptionHandler }
+            };
+
+            return handlers;
+        }
+
+        public override void Handle(ExceptionHandlerContext context)
+        {
+            var error = new ApiErrorModel(HttpStatusCode.InternalServerError);
+
+            AuthorizationLevel currentLevel = context.Request.GetAuthorizationLevel();
+            foreach (var handler in GetExceptionHandlers(context.Exception))
+            {
+                handler(context.ExceptionContext, currentLevel, error);
+            }
+
+            TraceErrorEvent(context.ExceptionContext, error);
+
+            context.Result = new ResponseMessageResult(context.Request.CreateResponse(error.StatusCode, error));
+        }
+
+        private IEnumerable<ExceptionProcessor> GetExceptionHandlers(Exception exc)
+        {
+            if (exc == null)
+            {
+                yield break;
+            }
+
+            // We return our default handler first.
+            // Subsequent handlers can override everything done by it.
+            yield return DefaultExceptionHandler;
+
+            Type exceptionType = exc.GetType();
+            ExceptionProcessor exceptionHandler = null;
+            if (exceptionType != null && _handlers.TryGetValue(exceptionType, out exceptionHandler))
+            {
+                yield return exceptionHandler;
+            }
+        }
+
+        private static void DefaultExceptionHandler(ExceptionContext exceptionContext, AuthorizationLevel currentLevel, ApiErrorModel error)
+        {
+            if (currentLevel == AuthorizationLevel.Admin || exceptionContext.RequestContext.IsLocal)
```

Comment:
Basically our version of `customErrors` as we don't allow web.config access.. unrelated to this codeflow, but could we support xdts for modifying the functions web.config? Can't remember if there are specific reasons why we would block that.